core/src/test/java/io/github/tcdl/msb/message/MessageFactoryTest.java

```diff
@@ -219,4 +220,32 @@ public void testCreateAckBuilder() throws Exception {
         Acknowledge ack = messageFactory.createAckBuilder().build();
         assertNotNull(ack.getResponderId());
     }
+
+    @Test
+    public void testCreateRequestMessageBuilderPublishedAtPresent() {
+        String bodyText = "body text";
+        Payload requestPayload = TestUtils.createPayloadWithTextBody(bodyText);
+
+        Builder requestMessageBuilder = TestUtils.createMessageBuilder(FIXED_CLOCK);
+
+        Message message = messageFactory.createRequestMessage(requestMessageBuilder, requestPayload);
+
+        assertNotNull(message.getMeta().getPublishedAt());
+        assertEquals(message.getMeta().getPublishedAt(), FIXED_CLOCK.instant());
+    }
+
+    @Test
+    public void testCreateRequestMessageBuilderPublishedAtIsAfterCreatedAt() {
+        String bodyText = "body text";
+        Payload requestPayload = TestUtils.createPayloadWithTextBody(bodyText);
+
+        Builder requestMessageBuilder = TestUtils.createMessageBuilder(null);
+
+        Message message = messageFactory.createRequestMessage(requestMessageBuilder, requestPayload);
```

Comment:
Unfortunately Java 8 doesn't have the suitable implementation of clock. Essentially what we need is a clock that returns an incremented value each time we call it. But we could implement it ourselves. Although I understand that it's far from scope of the current task.

For now please put a comment here why this test may blink.